compiler/src/org.graalvm.compiler.virtual/src/org/graalvm/compiler/virtual/phases/ea/GraphEffectList.java

```diff
@@ -173,6 +175,20 @@ public boolean isVisible() {
      */
     public void deleteNode(Node node) {
         add("delete fixed node", (graph, obsoleteNodes) -> {
+            if (node instanceof AccessFieldNode && node.hasUsages()) {
+                // see if it is associated with any membar nodes
+                // and if so remove the association
+                for (int i = 0; i < node.getUsageCount(); i++) {
```

Comment:
This is a bit unsafe while also implicitly modifying the usages: usages are not ordered in any way and can actually change order when a usage is added/removed (e.g., removing a usage usually swaps the last usage with the removed usage to avoid having to shift all usages after the removed one).

One way is to find what you want first and then modify thing. For example (note the `snapshot`):
```
for (MembarNode membar : node.usages().filter(MembarNode.class).snapshot()) {
    if (membar.getAccess() == node) {
        membar.setAccess(null);
        membar.setLeading(null);
    }
}
```

Reply:
Hi Gilles, I have pushed a fix to use your suggestion.